compiler/rustc_parse/src/parser/item.rs

```diff
@@ -2385,7 +2385,20 @@ impl<'a> Parser<'a> {
             };
             let (pat, ty) = if is_name_required || this.is_named_param() {
                 debug!("parse_param_general parse_pat (is_name_required:{})", is_name_required);
-
+                if let token::Ident(symname, _) = this.token.kind
+                    && this.prev_token.is_lifetime()
+                {
+                    // parse pattern with "'a Sized" we're supposed to give suggestion like
+                    // "'a + Sized"
+                    let mut err = this.struct_span_err(this.prev_token.span, &format!("expected `+` between lifetime and {}", symname.as_str()));
```

Comment:
This really shouldn't be creating a new error, but adding to an existing error.

Reply:
I moved the code to `parse_impl_ty ` which need to create a new error, because at that moment. There was no existing error to add.